mod.d/enadiag.yaml

```diff
@@ -0,0 +1,270 @@
+      return driver_version
+
+  def compare_ena_version():
+      """
+      Check to see if the ena version is new enough to have the necessary stats available.
+
+      Returns:
+          Boolean True if version is high enough
+      """
+      # Define version with stats included
+      stats_included = "2.2.10"
+      current_version = get_ena_version()
+      # Get only the version number and remove trailing letters to make version numbers comparable
+      comprable_version = re.search("(\d+.\d+.\d+)", current_version).group(0)
+      return LooseVersion(comprable_version) > LooseVersion(stats_included)
+
+  def get_ena_stats(interface):
+      """
+      Uses subprocess to pull ethtool statisics
+
+      Parameter:
+          interface (str): String with the interface (e.g. "eth0")
+
+      Returns
+          stat_output (str): String with full ethtool -S output.
+      """
+      try:
+          # ethtool -S output, will return all the stats we care about.
+          stat_output = subprocess.check_output(["ethtool", "-S", interface],
+                                                universal_newlines=True, stderr=subprocess.STDOUT)
+      except Exception as ex:
+          print(ex)
+          print("[WARN] Unable to get stats from ethool")
+      return stat_output
+
+  def save_ena_stats():
+      """
+      Regexes out the statistic values we care about and adds them to the Interface class object.
+      """
+      # Simple regex on each stat name and values, pull value group
+      try:
+          for interface in interface_results:
+              stat_output = get_ena_stats(interface.name)
+              interface.tx_timeout = re.search("(tx_timeout: )(\d+)", stat_output).group(2)
+              interface.suspend = re.search("(suspend: )(\d+)", stat_output).group(2)
+              interface.wd_expired = re.search("(wd_expired: )(\d+)", stat_output).group(2)
+              interface.interface_down = re.search("(interface_down: )(\d+)", stat_output).group(2)
+              interface.admin_q_pause = re.search("(admin_q_pause: )(\d+)", stat_output).group(2)
+              interface.bw_in_allowance_exceeded = re.search("(bw_in_allowance_exceeded: )(\d+)", stat_output).group(2)
+              interface.bw_out_allowance_exceeded = re.search("(bw_out_allowance_exceeded: )(\d+)", stat_output).group(2)
+              interface.pps_allowance_exceeded = re.search("(pps_allowance_exceeded: )(\d+)", stat_output).group(2)
+              interface.conntrack_allowance_exceeded = re.search("(conntrack_allowance_exceeded: )(\d+)",
+                                                                 stat_output).group(2)
+              interface.linklocal_allowance_exceeded = re.search("(linklocal_allowance_exceeded: )(\d+)",
+                                                                 stat_output).group(2)
+      except Exception as ex:
+          print(ex)
+          print("[WARN] Unable to save interface stats to class object.")
```

Comment:
- The warn message  is kind of ambiguous.
- Seems this would only happen if the ethtool output changes, right?
- Should the entire function fail if only one field is missing?

Reply:
Could we clarify the warn message then? Something along the lines of "expected ethtool output perhaps?

Reply:
Sure can!

Reply:
Can do.